__tests__/integration/template/transaction/template.test.ts

```diff
@@ -0,0 +1,346 @@
+import { GenesisWalletHelper } from '../../helpers/genesis-wallet.helper';
+import {
+  DEFAULT_PIN_CODE,
+  generateWalletHelper,
+  stopAllWallets,
+  waitForTxReceived,
+} from '../../helpers/wallet.helper';
+
+import HathorWallet from '../../../../src/new/wallet';
+import SendTransaction from '../../../../src/new/sendTransaction';
+import transactionUtils from '../../../../src/utils/transaction';
+import { TransactionTemplateBuilder } from '../../../../src/template/transaction/builder';
+import { WalletTxTemplateInterpreter } from '../../../../src/template/transaction/interpreter';
+import { TOKEN_AUTHORITY_MASK, TOKEN_MELT_MASK, TOKEN_MINT_MASK } from '../../../../src/constants';
+
+const DEBUG = true;
+
+describe('Template execution', () => {
+  let hWallet: HathorWallet;
+  let interpreter: WalletTxTemplateInterpreter;
+  let tokenUid: string;
+
+  beforeAll(async () => {
+    hWallet = await generateWalletHelper(null);
+    interpreter = new WalletTxTemplateInterpreter(hWallet);
+    const address = await hWallet.getAddressAtIndex(0);
+    await GenesisWalletHelper.injectFunds(hWallet, address, 10n, {});
+  });
+
+  afterAll(async () => {
+    await hWallet.stop();
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should be able to create a custom token', async () => {
+    const template = new TransactionTemplateBuilder()
+      .addConfigAction({ tokenName: 'Tmpl Test Token 01', tokenSymbol: 'TTT01' })
+      .addSetVarAction({ name: 'addr', call: { method: 'get_wallet_address' } })
+      .addUtxoSelect({ fill: 1 })
+      .addTokenOutput({ address: '{addr}', amount: 100, useCreatedToken: true })
+      .addAuthorityOutput({ authority: 'mint', address: '{addr}', useCreatedToken: true, count: 5 })
+      .addAuthorityOutput({ authority: 'melt', address: '{addr}', useCreatedToken: true, count: 5 })
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+    if (tx.hash === null) {
+      throw new Error('Transaction does not have a hash');
+    }
```

Comment:
suggestion: isn't it better to `expect`?

Reply:
Added expect, but need to keep the if statement to ensure typing.